### PR TITLE
Add complex type formatting

### PR DESCRIPTION
**Pull request by Jubeki**
This PR allows the formatting of Complex Types like Union Types or IntersectionTypes.

Examples for Complex Types:
```pho
Collection|string|null
Collection&AnotherClass
Collection<int, string>
array{'key': int}
Closure(string): int
```

I hope this are all cases which should be considered. But there can be more added later on if wanted.
I also added a simple unit test-suite for testing the formatting of complex types which runs on GitHub Actions if enabled.

### CODE REVIEW

**Comment by adamwojs** [comment]
Hi @Jubeki, thank you for contribution! It brings a lot o value to this package. Due to other duties I will need some time (1-2 days)  to  give you a more detailed feedback.

**Comment by Jubeki** [comment]
No Problem @adamwojs 
I just did some adjustments for performance reasons.

**Comment by Jubeki** [comment]
@adamwojs Do you have time to review the PR?

File: tests/CorrectlyFormatTypesTest.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+namespace AdamWojs\PhpCsFixerPhpdocForceFQCN\Tests;
+
+use AdamWojs\PhpCsFixerPhpdocForceFQCN\Analyzer\ImportInfo;
+use AdamWojs\PhpCsFixerPhpdocForceFQCN\Analyzer\NamespaceInfo;
+use AdamWojs\PhpCsFixerPhpdocForceFQCN\Analyzer\Range;
+use AdamWojs\PhpCsFixerPhpdocForceFQCN\FQCN\FQCNTypeNormalizer;
+use PhpCsFixer\Console\Application;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Output\BufferedOutput;
```
Comment (adamwojs) [inline tests/CorrectlyFormatTypesTest.php:13]:
```suggestion
```

Unused import

File: tests/CorrectlyFormatTypesTest.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
+<?php
+declare(strict_types=1);
+
+namespace AdamWojs\PhpCsFixerPhpdocForceFQCN\Tests;
+
+use AdamWojs\PhpCsFixerPhpdocForceFQCN\Analyzer\ImportInfo;
+use AdamWojs\PhpCsFixerPhpdocForceFQCN\Analyzer\NamespaceInfo;
+use AdamWojs\PhpCsFixerPhpdocForceFQCN\Analyzer\Range;
+use AdamWojs\PhpCsFixerPhpdocForceFQCN\FQCN\FQCNTypeNormalizer;
+use PhpCsFixer\Console\Application;
```
Comment (adamwojs) [inline tests/CorrectlyFormatTypesTest.php:10]:
```suggestion
```
Unused import

**Review by adamwojs — APPROVED** [review]

**Comment by adamwojs** [comment]
There is more code style issue in code base, so I will handle code review remarks on my own.  :wink:  

Thank you for contribution!

**Comment by Jubeki** [comment]
Thanks for formatting. 👍

**Pull request merged**